CHT/CHT.C

```diff
@@ -85,86 +85,26 @@ std::string preciceAdapter::CHT::ConjugateHeatTransfer::determineSolverType()
     // NOTE: When coupling a different variable, you may want to
     // add more cases here. Or you may provide the solverType in the config.
 
-    std::string solverType;
-
-    // Determine the solver type: Compressible, Incompressible or Basic.
-    // Look for the files transportProperties, turbulenceProperties,
-    // and thermophysicalProperties
-    bool transportPropertiesExists = false;
-    bool turbulencePropertiesExists = false;
-    bool thermophysicalPropertiesExists = false;
+    std::string solverType = "basic";
```

Comment:
> The type 'basic' is assumed instead of naming it 'other'. So we raise a warning and tell the user, that he might get issues with this settings. I preferred now 'basic', since this is an actual OpenFOAM type and the user might (or not) be fine with the settings. If we tell the user, that we calculate with type 'other', no one knows what this means without looking in the code.

What if we assume `unknown` here? The difference would be that `other` expects to read something from the config, while `unknown` will only appear if everything goes wrong.

> I preferred now 'basic', since this is an actual OpenFOAM type

Keep in mind that some users define their own solvers, which may be quite exotic (and complex, thus not `basic`). In OpenFOAM, `basic` is a small collection of solvers that do not have much in common that we can take advantage of. We could still support both `basic` and `other`, but then we cannot really distinguish between these two.

Reply:
> What if we assume unknown here? The difference would be that other expects to read something from the config, while unknown will only appear if everything goes wrong.

I'm fine with it. But what should the module do afterwards? Crash? Or select the 'basic' implementation?

Reply:
The workflow is now as follows:
Read from the dict: if (type specified) -> fine -> select appropriate type and calculate
                               else (determine) -> either we get inco or compr or unknown type-> how to proceed here

Reply:
If one of the two gets determined (i.e. there is a pressure field), let it be whatever gets determined. In any case, we should inform the user that we are determining the solver type automatically.

If none of these can be determined (i.e. there is no pressure field), we can either:
1. say "unknown solver type, please define it in preciceDict", or (not as important)
2. check if there is no pressure but there is temperature. In that case, we have our current "basic" type (and hopefully nothing else). If neither pressure not temperature is found, throw an error similar to (1).

Reply:
> we should inform the user that we are determining the solver type automatically.

We are informing the user [in a debug message](https://github.com/precice/openfoam-adapter/blob/11db960fdfa0534eaeeb847b402ececa8a8bcb68/CHT/CHT.C#L36)

Ok, then I vote for option one and I will throw an error